### ISSUE TITLE
Add a `kind` attribute to the `Talk` model 

### ISSUE THREAD

**Issue by adrienpoly** [adrienpoly/rubyvideo#325]
In the event model we have this query 

```ruby
def keynote_speakers
    talks.select { |talk|
      talk.title.start_with?("Keynote: ") ||
        talk.title.include?("Opening Keynote") ||
        talk.title.include?("Closing Keynote")
    }.flat_map(&:speakers)
  end
```

it would be great to replace it with a real db scope

**Comment by marcoroth** [comment]
Maybe it should be a `talk_type` since we also have other types like "Lightning Talk", "Panel", "Workshop", "Fishbowl" and potentially others.

**Comment by adrienpoly** [comment]
yes great idea 🎉

**Comment by nicogaldamez** [comment]
Hey @marcoroth @adrienpoly ! Happy to help with this.

I like the idea of adding an enum to `Talk` to indicate if it's a Keynote, Lightning Talk, Standard, etc. 

I’ve got a couple of questions:

1) What do you think about naming the enum `kind`? `type` would be ideal, but since it’s a reserved word, we’d have to prefix it, ending up with something like `talk_type`. I usually use `kind` in these cases.
2) We'll need to assign the enum for existing talks.  How are you handling data migrations right now?
3) I’m thinking we could also add this logic in `update_from_yml_metadata` to set the talk kind directly there

**Comment by marcoroth** [comment]
Hey @nicogaldamez, thanks, that would be awesome!

1. Yeah, I think `kind` is ideal since we already use in other models too 👍🏼 
2. Mostly we have been updating the YAML files in `data/` and then re-running the seeds.
3. Yeah, this is kinda what we have been doing, `update_from_yml_metadata!` is what we use to update the data in the database from the YAML files.

So either we can either teach the `update_from_yml_metadata!` what kind of talk it is based on the `title` attribute in the YAML file, or we backfill all the `kind`'s to the YAML files.

**Comment by nicogaldamez** [comment]
Thanks @marcoroth for the quick response :)

>Mostly we have been updating the YAML files in data/ and then re-running the seeds.

How does that work in production? Are seeds run with every deploy?

**Comment by marcoroth** [comment]
I think so, yes!

@adrienpoly has been deploying the changes, but that's how I designed the `update_from_yml_metadata` method to work.

**Comment by marcoroth** [comment]
Resolved via #334
